src/Sylius/Bundle/CoreBundle/PaymentRequest/CommandHandler/Payum/TokenPaymentRequestHandler.php

```diff
@@ -44,7 +44,9 @@ public function __invoke(PaymentRequestHashAwareInterface $command): void
         $request = $this->payumRequestFactory->createNewWithToken($token);
 
         $token = $request->getToken();
-        Assert::notNull($token, 'A Payum token cannot be null.');
+        if (null === $token) {
+            throw new NullPayumTokenException();
+        }
```

Comment:
As above, maybe there should be `PaymentTokenNotFoundException`. Also, I can't figure out why we assert if there's a token if we pass it at line `44`.

Reply:
yeah, but this method from payum could return null